src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,83 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+
 public class WorkWithFile {
+    private static final int OPERATION_TYPE_INDEX = 0;
+    private static final int AMOUNT_INDEX = 1;
+    private static final String SUPPLY_VALUE = "supply";
+    private static final String BUY_VALUE = "buy";
+    private static final String RESULT_VALUE = "result";
+    private static final String COMMA_VALUE = ",";
+    private static final String WHITESPACE = System.lineSeparator();
 
     public void getStatistic(String fromFileName, String toFileName) {
+        Path pathFromFile = Paths.get(fromFileName);
+        Path pathToFileName = Paths.get(toFileName);
+        String[] report = getReport(pathFromFile);
+        saveReport(report, pathToFileName);
+    }
+
+    private void createFile(Path path) {
+        try {
+            Files.createFile(path);
+        } catch (IOException e) {
+            throw new RuntimeException("Can't create file", e);
+        }
+    }
+
+    private void saveReport(String[] report, Path path) {
+        createFile(path);
+        for (String line : report) {
+            try {
+                Files.write(path, line.getBytes(), StandardOpenOption.WRITE);
+            } catch (IOException e) {
+                throw new RuntimeException("Can't write in file", e);
+            }
+        }
```

Comment:
with such implementation we will overwrite file each time when `Files.write` will be called
so maybe it would be better to write full text only once
try to use debugger and figure out what is going here (look at value of `line` variable at each iteration)